lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -718,18 +724,23 @@ def _stop_opensearch(self) -> None:
         # 3. Remove the exclusions
         self.opensearch_exclusions.delete_current()
 
-    def _restart_opensearch(self, event: EventBase) -> None:
+    def _restart_opensearch(self, event: _RestartOpenSearch) -> None:
         """Restart OpenSearch if possible."""
-        if not self.peers_data.get(Scope.UNIT, "starting", False):
-            try:
-                self._stop_opensearch()
-            except OpenSearchStopError as e:
-                logger.exception(e)
-                event.defer()
-                self.status.set(WaitingStatus(ServiceIsStopping))
-                return
+        if not self.node_lock.acquired:
+            logger.debug("Lock to restart opensearch not acquired. Will retry next event")
+            event.defer()
+            return
 
-        self._start_opensearch(event)
+        try:
+            self._stop_opensearch()
+        except OpenSearchStopError as e:
+            logger.exception(e)
+            # TODO: should lock be released?
+            event.defer()
```

Comment:
we can safely release release

Reply:
Yes - we should release, and it is safe because: the service would have failed to stop so it is not needed for the lock to be retained - a later retry (defer) should be enough.